File: blockchain_integration/pi_network/smartship/src/components/Transaction.js
import React, { useState, useEffect } from 'react';
import Web3 from 'web3';
import { useContractLoader, useContractReader } from 'eth-hooks';

const Transaction = ({ transactionId, provider, blockNumber }) => {
  const [sender, setSender] = useState('');
  const [receiver, setReceiver] = useState('');
  const [amount, setAmount] = useState(0);
  const contract = useContractLoader('MyContract', provider);

  useEffect(() => {
    async function fetchData() {
      const transaction = await contract.methods.transactions(transactionId).call();
      setSender(transaction.sender);
      setReceiver(transaction.receiver);
      setAmount(Web3.utils.fromWei(transaction.amount, 'ether'));
    }
    fetchData();
  }, [transactionId, blockNumber]);

  return (
    <div>
      <h2>Transaction: {transactionId}</h2>
      <p>Sender: {sender}</p>
      <p>Receiver: {receiver}</p>
      <p>Amount: {amount} ETH</p>
    </div>
  );
};

export default Transaction;

File: blockchain/core/transaction.py
from typing import List

class Transaction:
    def __init__(self, sender: str, receiver: str, amount: float):
        self.sender = sender
        self.receiver = receiver
        self.amount = amount

    def __str__(self):
        return f"Transaction - Sender: {self.sender}, Receiver: {self.receiver}, Amount: {self.amount}"

class TransactionPool:
    def __init__(self):
        self.transactions = []

    def add_transaction(self, transaction: Transaction):
        self.transactions.append(transaction)

    def get_transactions(self):
        return self.transactions
